### PR TITLE
Length of Longest Common Substring - String problem Two

### PR DESCRIPTION
**Pull request by Ganeshbabu18**

### CODE REVIEW

File: longest_common_substring.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+def longest_common_substring(X: str, Y: str) -> int:
+    m, n = len(X), len(Y)
```
Comment (amitsahu8) [inline longest_common_substring.py]: follow coding standard and guideline

File: longest_common_substring.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    table = [[0] * (n+1) for _ in range(m+1)]
```
Comment (amitsahu8) [inline longest_common_substring.py]: explain the logic

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: longest_common_substring.py (hunk 1 above)
Comment (Ganeshbabu18) [reply longest_common_substring.py]: I have made the changes. Kindly review it

File: longest_common_substring.py (hunk 2 above)
Comment (Ganeshbabu18) [reply longest_common_substring.py]: Initializing a matrix  and storing the length of the common substring

File: longest_common_substring.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+def longest_common_substring(str_one: str, str_two: str) -> int:
+    len_one, len_two = len(str_one), len(str_two)
+    table = [[0] * (len_one + 1) for _ in range(len_two + 1)]
```
Comment (amitsahu8) [inline longest_common_substring.py]: I didn't understood why you used table

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: longest_common_substring.py (hunk 3 above)
Comment (Ganeshbabu18) [reply longest_common_substring.py]: just a random matrix to store the lengths

Comment (amitsahu8) [reply longest_common_substring.py]: Resolve the conflicts

Comment (Ganeshbabu18) [reply longest_common_substring.py]: Conflict has been resolved, kindly check

**Review by amitsahu8 — APPROVED** [review]

**Pull request merged**